src/didactUriCompletionItemProviderAsciiDoc.ts

```diff
@@ -0,0 +1,78 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+'use strict';
+
+import * as vscode from 'vscode';
+import {didactProtocol, DidactUriCompletionItemProvider} from './didactUriCompletionItemProvider';
+import * as extensionFunctions from './extensionFunctions';
+
+export class DidactUriCompletionItemProviderAsciiDoc extends DidactUriCompletionItemProvider {
+
+	public provideCompletionItemsForDidactProtocol(text: string): vscode.CompletionItem[] {
+		let completions : vscode.CompletionItem[] = super.provideCompletionItemsForDidactProtocol(text);
+
+		let didactUri = this.getDidactUriFromLine(text);
+		if (didactUri) {
```

Comment:
could be combined into a single if statement

Reply:
I don't think so -- I did it and stashed it in the variable because I wanted to avoid doing the processing of the text twice